### PR TITLE
8339154: Cleanups and JUnit conversion of test/jdk/java/util/zip/Available.java

### PR DESCRIPTION
**Pull request by eirbjo**
Please review this test-only PR which addresses several issues with the `test/jdk/java/util/zip/Available.java` test:

* The test is converted to JUnit 5
* The test now creates its own test vector programmatically instead of relying on a binary `input.jar` test vector 
* Coverage is added for calling `available()` after calling `ZipInputStream.closeEntry`, as expected by the API specification for `ZipInputStream.available`
* Coverage is added for calling `available()` on a closed `ZipInputStream`
* Coverage is added for the unspecified, but long-standing behavior of `ZipFileInputStream.available()` (The InputStream returned for `STORED` entries)

Additionally, the test is split into multiple methods, adding javadoc comments for each of them.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8339154](https://bugs.openjdk.org/browse/JDK-8339154): Cleanups and JUnit conversion of test/jdk/java/util/zip/Available.java (**Enhancement** - P5)

### Reviewers
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20744/head:pull/20744` \
`$ git checkout pull/20744`

Update a local copy of the PR: \
`$ git checkout pull/20744` \
`$ git pull https://git.openjdk.org/jdk.git pull/20744/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20744`

View PR using the GUI difftool: \
`$ git pr show -t 20744`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20744.diff">https://git.openjdk.org/jdk/pull/20744.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20744#issuecomment-2315117266)

### CODE REVIEW

File: test/jdk/java/util/zip/Available.java [hunk 1]
```diff
@@ -22,69 +22,162 @@
...
+     * This verifies unspecified, but long-standing behavior. See 4401122.
+     *
+     * @throws IOException if an unexpected error occurs
+     */
+    @Test
+    public void testZipFileInflaterInputStream() throws IOException {
+        try (ZipFile zfile = new ZipFile(zip.toFile())) {
+            assertRemainingUncompressedBytes(zfile, "deflated.txt");
         }
     }
```
Comment (LanceAndersen) [inline test/jdk/java/util/zip/Available.java:147]: Could we collapse the two above tests  via

```
@ParameterizedTest
@ValueSource(strings = { "stored.txt", "delated.txt" })
void testAvailbleRemainingBytes(iString zipEntry) {
    try (ZipFile zfile = new ZipFile(zip.toFile())) {
            assertRemainingUncompressedBytes(zfile, zipEntry);
        }
}
```

Comment (eirbjo) [reply test/jdk/java/util/zip/Available.java:135]: I like it!

I collapsed `ZipFile` related tests into the method `testZipFileStreamsRemainingBytes` and inlined the `assertRemainingUncompressedBytes` method into that.

A few minor improvements to the inlined `assertRemainingUncompressedBytes` code body:

* Added a comment saying that the `InputStream` could be `ZipFileInputStream` or `ZipFileInflaterInputStream`
* Moved the "decrement by one" assertion logic into the loop, asserting decrement on each iteration 
* Cleaned up some spacing issues in the for loop

**Review by LanceAndersen — APPROVED** [review]
Thank you for the additional tweaks/improvements.


Test looks much better

**Comment by eirbjo** [comment]
Turns out that naming a ZipEntry "stored.txt" does not actually make it stored! 🤦 

@LanceAndersen I'll need a re-review for the following line in `setup`:

`stored.setMethod(ZipEntry.STORED);`

I have verified using the debugger that each invocation now sees the expected ZipFile InputStream implementation.

**Review by LanceAndersen — APPROVED** [review]

**Pull request closed without merging**